storage/rocksdb/rdb_datadic.cc

```diff
@@ -780,6 +783,53 @@ int Rdb_key_def::successor(uchar *const packed_tuple, const uint &len) {
   return changed;
 }
 
+bool Rdb_key_def::covers_lookup(TABLE *const table,
+                                const rocksdb::Slice *const unpack_info) const {
+  DBUG_ASSERT(m_index_type == INDEX_TYPE_SECONDARY);
+  DBUG_ASSERT(table->key_info != nullptr);
+  Rdb_string_reader unp_reader = Rdb_string_reader::read_or_empty(unpack_info);
+
+  // Check if unpack_info is corrupted
+  const bool has_unpack_info =
+      unp_reader.remaining_bytes() &&
+      *unp_reader.get_current_ptr() == RDB_UNPACK_DATA_TAG;
+  if (!has_unpack_info || !unp_reader.read(RDB_UNPACK_HEADER_SIZE)) {
+    return false;
+  }
+
+  MY_BITMAP covered_bitmap;
+  my_bitmap_map covered_bits;
+  bitmap_init(&covered_bitmap, &covered_bits, 16, false);
+  if (unp_reader.read_uint16(&covered_bits)) {
+    return false;
+  }
+
+  uint covered_columns = 0;
+  KEY key_info = table->key_info[get_keyno()];
+  for (uint i = 0; i < m_key_parts; i++) {
+    uint fieldnr;
+    if (i < key_info.actual_key_parts) {
+      fieldnr = key_info.key_part[i].fieldnr;
+    } else if (!table_has_hidden_pk(table)) {
+      uint j = i - key_info.actual_key_parts;
+      fieldnr = table->key_info[table->s->primary_key].key_part[j].fieldnr;
+    } else {
+      break;
+    }
+
+    // Checks two conditions:
+    // - The column corresponding with this key part was requested
+    // - The entire column value is contained in the SK
+    if (bitmap_is_set(table->read_set, fieldnr - 1) &&
+        bitmap_is_set(&covered_bitmap, i)) {
```

Comment:
If the requested column is not covered, this can probably return false immediately.